File: backend/app/schemas/order.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

class DynamicMeasurementInput(BaseModel):
    field_id: int
    value: float

class OrderCreate(BaseModel):
    customer_id: int
    garment_type: str
    occasion: Optional[str] = None
    due_date: Optional[datetime] = None
    priority: Optional[str] = "Medium"
    tailor_remarks: Optional[str] = None
    customer_instructions: Optional[str] = None
    measurements: Optional[List[DynamicMeasurementInput]] = None
    staff_id: Optional[int] = None

class OrderRead(BaseModel):
    id: int
    customer_id: int
    garment_type: str
    occasion: Optional[str] = None
    status: str
    priority: Optional[str] = "Medium"
    due_date: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: Optional[datetime] = None
    tailor_remarks: Optional[str] = None
    customer_instructions: Optional[str] = None
    customer_name: Optional[str] = None
    customer_phone: Optional[str] = None
    staff_id: Optional[int] = None
    staff_name: Optional[str] = None
    measurements: List[dict] = []

    class Config:
        orm_mode = True

class OrderUpdate(BaseModel):
    status: Optional[str] = None
    priority: Optional[str] = None
    due_date: Optional[datetime] = None
    tailor_remarks: Optional[str] = None
    customer_instructions: Optional[str] = None
    staff_id: Optional[int] = None

File: frontend/flutter_app/lib/features/orders/models/order.dart
class Order {
  final int? id;
  final int customerId;
  final String garmentType;
  final String? occasion;
  final String? status;
  final String? priority;
  final String? dueDate;
  final String? completedAt;
  final String? createdAt;
  final String? tailorRemarks;
  final String? customerInstructions;
  final String? customerName;
  final String? customerPhone;
  final List<dynamic>? measurements;

  Order({
    this.id,
    required this.customerId,
    required this.garmentType,
    this.occasion,
    this.status,
    this.priority,
    this.dueDate,
    this.completedAt,
    this.createdAt,
    this.tailorRemarks,
    this.customerInstructions,
    this.customerName,
    this.customerPhone,
    this.measurements,
  });

  factory Order.fromJson(Map<String, dynamic> json) => Order(
        id: json['id'],
        customerId: json['customer_id'] ?? 0,
        garmentType: json['garment_type'] ?? '',
        occasion: json['occasion'],
        status: json['status'],
        priority: json['priority'],
        dueDate: json['due_date'],
        completedAt: json['completed_at'],
        createdAt: json['created_at'],
        tailorRemarks: json['tailor_remarks'],
        customerInstructions: json['customer_instructions'],
        customerName: json['customer_name'],
        customerPhone: json['customer_phone'],
        measurements: json['measurements'],
      );
}
